gallery_dl/extractor/postmill.py

```diff
@@ -0,0 +1,204 @@
+        self.save_link_post_body = self.config("save-link-post-body", False)
+        self._search_canonical_url = re.compile(r"/f/([\w\d_]+)/(\d+)/").search
+        self._search_image_tag = re.compile(
+            r'<a href="[^"]+"\n +class="submission__image-link"').search
+
+    def items(self):
+        for post_url in self.post_urls():
+            response = self.request(post_url)
+            extr = text.extract_from(response.text)
+
+            title = text.unescape(extr(
+                '<meta property="og:title" content="', '">'))
+            date = text.parse_datetime(extr(
+                '<meta property="og:article:published_time" content="', '">'))
+            username = extr(
+                '<meta property="og:article:author" content="', '">')
+            post_canonical_url = text.unescape(extr(
+                '<link rel="canonical" href="', '">'))
+
+            url = text.unescape(extr(
+                '<h1 class="submission__title unheaderize inline"><a href="',
+                '"'))
+            body = extr(
+                '<div class="submission__body break-text text-flow">',
+                '</div>')
+
+            match = self._search_canonical_url(post_canonical_url)
+            forum = match.group(1)
+            id = int(match.group(2))
+
+            is_text_post = url.startswith("/")
+            is_image_post = self._search_image_tag(response.text) is not None
+            data = {
+                "title": title,
+                "date": date,
+                "username": username,
+                "forum": forum,
+                "id": id,
+                "flair": [text.unescape(i) for i in text.extract_iter(
+                    response.text, '<span class="flair__label">', '</span>')],
+                "instance": self.instance,
+            }
+
+            urls = []
+            if is_text_post or self.save_link_post_body:
+                urls.append((Message.Url, "text:" + body))
+
+            if is_image_post:
+                urls.append((Message.Url, url))
+            elif not is_text_post:
+                urls.append((Message.Queue, url))
+
+            data["count"] = len(urls)
+            yield Message.Directory, data
+            for data["num"], (msg, url) in enumerate(urls, 1):
+                if url.startswith("text:"):
+                    data["filename"], data["extension"] = "", "htm"
+                else:
+                    data = text.nameext_from_url(url, data)
+
+                yield msg, url, data
+
+
+class PostmillSubmissionsExtractor(PostmillExtractor):
+    """Base class for Postmill submissions extractors"""
+    whitelisted_parameters = ()
+
+    def __init__(self, match):
+        PostmillExtractor.__init__(self, match)
+        self.base = match.group(3)
+        self.sorting_path = match.group(4) or ""
+        self.query = {key: value for key, value in text.parse_query(
+            match.group(5) or "").items() if self.acceptable_query(key)}
```

Comment:
Using `match.group()` with a static index does not work for BaseExtractor classes. The actual index shifts forward with each added instance.

Use `match.group(match.lastindex)` like for example [here](https://github.com/mikf/gallery-dl/blob/01bb75f6cbb108279c50c2a89de5c76ef677429b/gallery_dl/extractor/foolfuuka.py#L115-L116).

Reply:
And from personal experience: `match.lastindex` works "weird" with optional capture groups at the end of a pattern. It might be necessary to add an empty, non-optional capture group after each `QUERY_RE` to have consistent `lastindex` values.